msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignUpTest.kt

```diff
@@ -60,31 +83,20 @@ class SignUpTest : NativeAuthPublicClientApplicationAbstractTest() {
      * Running with runBlocking to avoid default 10 second execution timeout.
      */
     @Test
-    fun testSignUpSuccessSimple() = runBlocking {
-        var retryCount = 0
+    fun testSignUpSuccessSimple() {
         var signUpResult: SignUpResult
         var otp: String
-        var shouldRetry = true
 
-        while (shouldRetry) {
-            try {
+        retryOperation {
+            runBlocking {
                 val user = tempEmailApi.generateRandomEmailAddress()
                 val password = getSafePassword()
                 signUpResult = application.signUp(user, password.toCharArray())
                 Assert.assertTrue(signUpResult is SignUpResult.CodeRequired)
                 otp = tempEmailApi.retrieveCodeFromInbox(user)
                 val submitCodeResult = (signUpResult as SignUpResult.CodeRequired).nextState.submitCode(otp)
                 Assert.assertTrue(submitCodeResult is SignUpResult.Complete)
-                shouldRetry = false
-                break
-            } catch (e: IllegalStateException) {
-                // Re-run this test if the OTP retrieval fails. 1SecMail is known for emails to sometimes never arrive.
```

Comment:
Can we leave the comment, somewhere in the test? I feel like it's important context for (future) devs to know about.